### PR TITLE
feat: add dummy implementation of `pretouch()` for `!amd64` arch

### PR DESCRIPTION
**Pull request by xiekeyi98**
✅ Closes: #301

### CODE REVIEW

**Comment by xiekeyi98** [comment]
@AsterDY  Would you mind taking a look? Thanks!

File: sonic_!amd64.go [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package sonic
+
+import (
+	"reflect"
```
Comment (AsterDY) [inline sonic_!amd64.go:22]: please use the same format with other sonic codes:
- single quote import package
- four space as indent

File: sonic_!amd64_test.go [hunk 2]
```diff
@@ -0,0 +1,40 @@
+//go:build !amd64
```
Comment (AsterDY) [inline sonic_!amd64_test.go:1]: old version of go build constraint is also need, to be compatible with old go version

File: sonic_!amd64.go [hunk 3]
```diff
@@ -0,0 +1,36 @@
+//go:build !amd64
```
Comment (AsterDY) [inline sonic_!amd64.go:1]: old version of go build constraint is also need, to be compatible with old go version

**Review by AsterDY — COMMENTED** [review]
don't add more go files. just use compat.go and api_test.go is enough

File: sonic_!amd64.go (hunk 1 above)
Comment (xiekeyi98) [reply sonic_!amd64.go:22]: Okay.
I was wondering why we do not use `gofmt` to format golang code.

Is there a script tool or shell script to automatically format it?

**Review by AsterDY — APPROVED** [review]

**Pull request merged**